test-resources/headless-core-stub/lambdas/callback/tests/callback-handler.test.ts

```diff
@@ -71,18 +64,92 @@ describe("callback-handler", () => {
         });
     });
 
-    it("returns 404 if session is not found", async () => {
-        sessionByAuthorizationCodeSpy = jest
-            .spyOn(CallBackService.prototype, "getSessionByAuthorizationCode")
-            .mockRejectedValueOnce(new Error("No session item found for provided authorizationCode"));
+    it("uses client_id from request params if provided", async () => {
+        const clientIdOverride = "test-client-id";
 
-        const event: Partial<APIGatewayProxyEvent> = { queryStringParameters: { code: authorizationCode } };
-        const response = await lambdaHandler(event as APIGatewayProxyEvent, {} as Context);
+        jest.spyOn(KeyJwtHelper, "generatePrivateJwtParams").mockResolvedValueOnce(keyJwtValue);
+
+        getParametersSpy = jest.spyOn(ClientConfiguration, "getConfig").mockResolvedValueOnce({
+            redirectUri,
+            audience,
+            issuer: "https://issuer.example.com",
+            privateSigningKey: JSON.stringify({}),
+        });
+        getTokenSpy = jest.spyOn(CallBackService.prototype, "invokeTokenEndpoint").mockResolvedValueOnce({
+            statusCode: 200,
+            body: JSON.stringify({ access_token: accessTokenValue }),
+        });
+        issueCredentialSpy = jest.spyOn(CallBackService.prototype, "invokeCredentialEndpoint").mockResolvedValueOnce({
+            statusCode: 200,
+            body: "vc.jwt.credential",
+        });
 
-        expect(sessionByAuthorizationCodeSpy).toHaveBeenCalledWith(sessionTableName, authorizationCode);
+        const response = await lambdaHandler(
+            {
+                queryStringParameters: { code: authorizationCode, client_id: clientIdOverride },
+            } as unknown as APIGatewayProxyEvent,
+            {} as Context,
+        );
+
+        expect(getParametersSpy).toHaveBeenCalledWith(clientIdOverride);
+        expect(getTokenSpy).toHaveBeenCalledWith("my-audience/token", keyJwtValue);
+        expect(issueCredentialSpy).toHaveBeenCalledWith("my-audience/credential/issue", "access_token_value");
         expect(response).toEqual({
-            statusCode: 500,
-            body: "No session item found for provided authorizationCode",
+            statusCode: 200,
+            headers: {
+                "Content-Type": "text/plain",
+            },
+            body: "vc.jwt.credential",
+        });
+    });
+
+    it("uses state from request params if provided", async () => {
```

Comment:
nice, how about when this state is not provided? would be very nice to have a test for that

Reply:
That would be covered in the `returns 200 when entire flow is successful` test